lib/corpus.py

```diff
@@ -25,6 +25,9 @@ def __init__(self):
         super().__init__()
         self.name: Optional[str] = None
 
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self.fullname()})"
```

Comment:
I don't understand. There is no `fullname` method here? Maybe you want to add that as an abstract base method?

Reply:
The class `Speaker` inherits from `NamedEntity` and does not have a fullname - implementation wise and also conceptually. 

> add that as an abstract base method?

or even a concrete method that returns `self.name or ""` and then all Corpus/Subcorpus/Recordings/Segments already have their own implementations that give the full name.

Reply:
Right, I missed that. So should we just do `self.name or ""` here? I could then also add `__repr__` for `Corpus` and `Recording` using `fullname` there.

Reply:
I have no preference as long as it is correct in all cases :)